### PR TITLE
Windows: Factor out sqlite

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard <jhoward@microsoft.com>

Based on https://github.com/docker/docker/pull/28964#issuecomment-263929569, it's true :smile:. SQLite dependency on Windows is removed.  

@thaJeztah If we can get this into 1.13, I can update #28964 to match.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Just discussed this, but we don't think it's needed for 1.13, but definitely nice for master/next release

**Comment by cpuguy83** [comment]
Don't need it at all for 1.14, I think, just need to ensure we tell users "You need to upgrade from a previous version first" if we find a container with old links.

**Comment by jhowardmsft** [comment]
> I think, just need to ensure we tell users "You need to upgrade from a previous version first" if we find a container with old links.

@cpuguy83 How is that relevant to docker on Windows? This migration pulls in SQLite3 which is in-necessary on Windows.

**Comment by justincormack** [comment]
If we are going to remove sqlite completely in 1.14 then we dont need this I guess. Shall we just rip it out now?

File: daemon/links_unix.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
+// +build !windows
```
Comment (justincormack) [inline daemon/links_unix.go]: this is only needed on linux, no other platform (eg solaris) was supported then either.

**Review by justincormack — CHANGES_REQUESTED** [review]

**Comment by justincormack** [comment]
I would be inclined to put this into 1.13 though.

**Review by cpuguy83 — APPROVED** [review]
LGTM after changing `_unix` to `_linux` as @justincormack suggested.

**Comment by cpuguy83** [comment]
@jhowardmsft You're right, it's helpful to have the split (since Windows doesn't need to error out) even though we want to remove it on Linux too.

**Comment by jhowardmsft** [comment]
@justincormack @cpuguy83 Updated for Linux only. 

@thaJeztah I do think this does make sense though for 1.13 and the first official Windows release. But will leave that for you and @vieux to decide on.

**Comment by thaJeztah** [comment]
Just re-discussed this with @vieux, and he's ok with doing it in 1.13

**Comment by jhowardmsft** [comment]
@justincormack Just needs you to approve the _Linux updates :)

**Review by justincormack — APPROVED** [review]

**Pull request merged**